### DISCUSSION TITLE
Use a Surge CDN specific region server

### DISCUSSION THREAD

**Discussion by fu-sen** [sintaxi/surge#444] [Ideas]
If you want to use Surge with your own domain, point to `45.55.110.124`(A record) or `na-west1.surge.sh`(CNAME record):
https://surge.sh/help/adding-a-custom-domain

You can instead choose a server located in a specific region of the Surge CDN:
https://gist.github.com/sintaxi/670654550d4b30354614c7cfb72d5bf5
https://twitter.com/surge_sh/status/986960078078013441

If `na-west1.surge.sh` hasn't selected the right server in your area, or if the server is down, you'll find this switch useful.

**Comment by sintaxi** [comment]
The surge platform is in the process of moving over to a new DNS system. `geo.surge.world` is the geo-aware CNAME record moving forward. `na-west1.surge.sh` will become will automatically switch over and become legacy soon and everything is ported to the new system. `surge.world` provides both NS servers with full DNS management as well as the `geo.surge.world` record for those who want to point to the surge network using their own DNS management.

You can use `geo.surge.world` now if you like. If it doesn't direct you to the most optimal datacenter please report your findings. Instructions and information for the new DNS system will be found at https://surge.world

<img width="1192" alt="Screen Shot 2021-04-04 at 1 36 18 PM" src="https://user-images.githubusercontent.com/867/113520964-5240db00-954b-11eb-922e-46c55fd88c45.png">

**Reply by fu-sen** [reply]
I got information from `geo.sorge.world` a while back and tested it. It chooses the most ideal server, especially in some regions such as Asia. Good job! 👍

**Reply by fu-sen** [reply]
A new name server has been added to the announcement.
There's one thing you care about, but I'll find the answer in the development version:
```
PS C:\Users\balloon> surge --help

  Surge.sh - Static Web Publishing 0.23.0-rc.4

  surge                             (opt --preview)  publish with prompts
  surge <path>   <domain>           (opt --preview)  publish without prompts (recommended)
  surge encrypt  <domain>                            provision SSL cert for project
  surge certs    <domain>                            view certs for project
  surge config   <domain>                            view/change project configuration
  surge list     <domain>                            list all project revisions
  surge rollfore <domain>                            change to next revision
  surge rollback <domain>                            change to previous revision
  surge cutover  <domain>                            change to latest revision
  surge discard  <revision>                          remove revision from system
  surge files    <domain>                            list all project files
  surge audit    <domain>                            audit edgenode state
  surge bust     <domain>                            busts cache on all edgenodes
  surge traffic  <domain>                            analytics showing project traffic
  surge audience <domain>                            analytics showing audience device info
  surge usage    <domain>                            analytics showing bandwidth usage
  surge teardown <domain>                            tear down a published project
  surge ssl      <domain> <pempath>                  legacy command for uploding .pem file
  surge invite   <domain> <emails>                   invites user to be a contributor
  surge revoke   <domain> <emails>                   revokes contributor rights
  surge dns      <domain>                            view DNS records
  surge dns      <domain> add <type> <name> <value>  add DNS record
  surge dns      <domain> rem <id>                   remove DNS record
  surge whoami                                       show who you are logged in as
  surge login                                        only performs authentication step
  surge logout                                       expire local token
  surge token                                        create token for automation purposes
  surge plan                                         upgrade or downgrade account plan
  surge nuke                                         permanently removes account
  surge list                                         list all projects
  surge --version                                    outputs version
  surge --help                                       outputs this help message

PS C:\Users\balloon> ▄
```
We will wait a little longer to get it. ☕

**Comment by fu-sen** [comment]
Yeah， It looks like you've also applied `geo.surge.world` to the currently published` SUBDOMAIN.surge.sh` and `na-west1.surge.sh`.
As a result, I confirmed that the server in Singapore was referenced by me in Japan.

```
PS C:\Users\balloon> dig sgp.surge.sh

;; ANSWER SECTION:
sgp.surge.sh.           301     IN      A       139.59.195.30

PS C:\Users\balloon> dig na-west1.surge.sh

;; ANSWER SECTION:
na-west1.surge.sh.      301     IN      A       139.59.195.30

PS C:\Users\balloon> dig surge.sh

;; ANSWER SECTION:
surge.sh.               301     IN      A       139.59.195.30

PS C:\Users\balloon> ▄
```

I remember referring to New York until recently. Obviously the response should have been faster in some areas!
